orchest/orchest
# Improve analytics data structure

## Description

This PR introduces changes to our analytics module so that all data passes through the same logic, ensuring that we can easily work with the data thanks to the common structure. Additionally, the new implementation provides (inside the `_Anonymizer` class) an extensive overview of all the events that we support.

Structure always in the following format:
```
        {
            "event_properties": ...  # anonymized event properties
            "derived_properties": ...  # derived from removed properties
            "app_properties": ...  # Orchest application properties
        }
```

### How to test
1. `./orchest start --log-level=DEBUG`
2. Trigger some events
3. Check the `docker logs orchest-webserver` and note entries such as `Successfully sent analytics event 'session start'`.

In addition I checked the PostHog data to make sure entries are created.
